src/backchannel_logout.rs

```diff
@@ -0,0 +1,469 @@
+//! This module implements components needed for [back-channel logout]
+//!
+//! [back-channel logout]: <https://openid.net/specs/openid-connect-backchannel-1_0.html>
+
+use std::collections::HashMap;
+
+use chrono::{DateTime, Utc};
+use serde::{de::Error, Deserialize};
+
+use crate::{
+    types::helpers::{deserialize_string_or_vec, serde_utc_seconds},
+    types::SessionIdentifier,
+    Audience, IssuerUrl, SubjectIdentifier,
+};
+
+/// The Logout Token as defined in [section 2.4] of the [OpenID Connect Back-Channel Logout spec][1]
+///
+/// [section 2.4]: <https://openid.net/specs/openid-connect-backchannel-1_0.html#LogoutToken>
+/// [1]: <https://openid.net/specs/openid-connect-backchannel-1_0.html>
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct LogoutToken {
+    /// The issuer of this token
+    iss: IssuerUrl,
+    /// The audience this token is intended for
+    aud: Vec<Audience>,
+    /// Time at which this token was issued
+    iat: DateTime<Utc>,
+    /// The unique identifier for this token. This can be used to detect
+    /// replay attacks.
+    jti: String,
+    identifier: Identifier,
+    events: HashMap<String, serde_json::Value>,
+}
```

Comment:
Let's add a flattened generic `additional_claims` struct here too, similar to what the other token types do. This will allow users to define their own extensions. The spec says:
> Logout Tokens MAY contain other Claims. Any Claims used that are not understood MUST be ignored.

As long as the user's code understands the additional fields, then I think it's fine for them to be present.

I think this will need to use `FilteredFlatten` (see `IdTokenClaims::additional_claims` for an example) to filter out `sid` and `sub` since we're using multiple flattened fields and `Identifier`'s field names aren't known to serde's derive macro at compile time).

Reply:
Hopefully it sounds like more of a pain that it'll actually be to implement. I've added various utility types (e.g., `FilteredFlatten`) for doing this in other places without much new code each time. This will be the third instance (after ID tokens and user info responses) of having extensible JWTs.